server/sockets/socketApi.js

```diff
@@ -0,0 +1,93 @@
+var socket_io = require("socket.io");
+const verifyToken = require("../helpers/verifyToken");
+var io = socket_io();
+var socketApi = {};
+
+const roomMap = new Map();
+const codeMap = new Map();
+socketApi.io = io;
+
+io.on("connection", (socket) => {
+  console.log("connected");
+  socket.on("joinInterviewLobby", function (info, fn) {
+    let tempUsers = roomMap.get(info.id);
+    let createdUser = {};
+    if (tempUsers) {
+      if (tempUsers.userNames.indexOf(info.name) != -1) {
+        //user is not added since they are already in the room list
+      } else if (tempUsers.userNames.length > 1) {
+        fn(false);
+        return "too many users";
+      } else {
+        createdUser.userNames = [...tempUsers.userNames, info.name];
+        createdUser.userIds = [...tempUsers.userIds, info.userId];
+        tempUsers = createdUser;
```

Comment:
Can we clean this up a bit? For example, I don't think we need to create a new object of createdUser, we can just modify tempUsers directly and also we could slightly change the format of tempUsers to make it easier. For example, what about below:

roomMap[info.id] could have one array of users, where each index is an object

```
const tempUsers = roomMap.get(info.id); 

if (tempUsers.users.length > 1) {
  fn(false);
  return "too many users";
} else if (!tempUsers.users.find(user => user.id === info.userId)) {
   tempUsers.users.push({name: info.name, id: info.userId});
}
```